### ISSUE TITLE
False negative: Link must have discernible text

### ISSUE THREAD

**Issue by Brynman** [dequelabs/axe-core#4709]
### Product

axe Extension

### Product Version

4.10.2

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation
I expect a link with its name coming from an alt within a nested `<picture>` element to pass `Ensure links have discernible text`  

#### Actual
The `a` is flagged as having no link text.

#### How to Reproduce
Open Chrome extension and run on this page https://www.argos.co.uk/product/2079998?clickCSR=slp:cannedSearch

#### Additional context

<img width="1280" alt="Image" src="https://github.com/user-attachments/assets/8a172ddf-8c7d-410f-9f85-0eb28fe14af3" />
<img width="1280" alt="Image" src="https://github.com/user-attachments/assets/5728165a-9b4b-4e17-9710-ddc176e3de3b" />

**Comment by straker** [comment]
Thanks for the issue. I took a look at this and what I believe is happening is that the anchor in question is not fully loaded when the test is being run. When the page is first loaded the anchor is offscreen and appears to be empty, causing the anchor to be flagged as a violation.

```html
<a href="https://apps.apple.com/gb/app/argos/id370371087" target="_blank" rel="noreferrer" data-test="footer-app-apple" style="margin-right: 16px;">
  <span></span>
</a>
```

However when the page is scrolled to the bottom the anchor is then populated with the App Store image. At that point the anchor gets an accessible name from the picture element and does not trigger a violation.

Since the anchor is present on the page before it is populated with an accessible name, axe is correctly flagging this as an issue (since screen readers can access the anchor before it becomes populated).
